doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +558,149 @@ include the chapel runtime and standard modules for use in a non-Chapel program
 and when the library is linked to a Chapel program this leads to multiple
 definitions of these functions.
 
+Using Your Library in Multilocale Settings
+==========================================
+
+Prerequisites
+-------------
+
+Chapel supports ``--library`` when ``CHPL_COMM = gasnet``.  We intend to support
+other settings in the future, see :ref:`Other Settings` in the :ref:`Multilocale
+Caveats` section for more information.
+
+To compile a multilocale library, `ZeroMQ <http://zeromq.org/>`_ must be
+installed.
+
+If ZeroMQ is not installed in a way that enables your C compiler to find it
+easily, the environment variable ``CHPL_ZMQ_HOME`` can be set.  This environment
+variable should be set to a directory containing both an ``include`` directory
+which contains ``zmq.h`` and a ``lib`` directory which contains ``libzmq.*``.
+For example, for a directory structure:
+
+.. code-block:: text
+
+   |-- .local/
+   |    |-- include/
+   |    |    |-- zmq.h
+   |    |-- lib/
+   |    |    |-- libzmq.a
+   |    |    |-- libzmq.so
+
+``CHPL_ZMQ_HOME`` would be set to ``/absolute/path/to/.local/``.
+
+Initializing Your Multilocale Library
+-------------------------------------
+
+Multilocale libraries can be used in a manner similar to single locale
+libraries.  However, as with transitioning between a single locale executable
+and a multilocale one, it is necessary to specify the number of locales required
+for the multilocale library.
+
+In C
+~~~~
+
+Users must still call ``chpl_library_init()`` before utilizing the exported
+Chapel functions.  However, the ``char* argv[]`` must now include two additional
+entries: the numlocales flag and its intended value.
+
+This can be accomplished either by explicitly adding the arguments in the C
+client program itself, or by passing them as arguments to the executable.
+
+This example demonstrates explicitly adding the arguments in the program using
+the ``foo`` library.
+
+.. code-block:: C
+
+   #include "foo.h"
+
+   int main(int argc, char* argv[]) {
+     int argChapelC = 3;
+     char* argChapelV[3] = {argv[0], "-nl", "2"};
+     // Initialize the Chapel runtime and standard modules
+     chpl_library_init(argChapelC, argChapelV);
+
+     baz(7); // Call into a library function
+
+     chpl_library_finalize();
+
+     return 0;
+   }
+
+Alternatively, the original single locale client from `Initializing Your
+Library In C`_ can be used with the additional two arguments to the executable:
+
+.. code-block:: bash
+
+   ./a.out -nl 2
+
+Users also still need to call the generated module initialization function for
+multilocale libraries, as mentioned in that section.
+
+In Python
+~~~~~~~~~
+
+Users must still call ``chpl_setup()`` before utilizing the exported Chapel
+functions.  However, it requires a ``numLocales`` argument when the library
+has been compiled for multilocale settings.  E.g. to run with ``4`` locales,
+write ``chpl_setup(4)``.
```

Comment:
nit: Putting this `chpl_setup(4)` in a code block would make it stand out more for skimming readers, e.g.

---

The example below sets up the Chapel runtime to run with ``4`` locales:

```python
chpl_setup(4)
```